### PR TITLE
Always retry on NuGet cache corruptions to reduce its impact

### PR DESCRIPTION
**Pull request by smile21prc**
Always retry on NuGet cache corruptions to reduce its impact. until it's
root caused and fixed.

### CODE REVIEW

**Comment by smile21prc** [comment]
@mmitche @jaredpar , can you take a look? Thanks.

File: jobs/generation/Utilities.groovy [hunk 1]
```diff
@@ -637,7 +637,9 @@ class Utilities {
             '\'type_traits\' file not found', // This is here for certain flavors of clang on Ubuntu, which can exhibit odd errors.
             '\'typeinfo\' file not found', // This is here for certain flavors of clang on Ubuntu, which can exhibit odd errors.
             'Only AMD64 and I386 are supported', // Appears to be a flaky CMAKE failure
-            'java.util.concurrent.ExecutionException: Invalid object ID'
+            'java.util.concurrent.ExecutionException: Invalid object ID',
+            '.*XmlException.', // This is here until NuGet cache corruption issue is root caused and fixed.
```
Comment (mmitche) [inline jobs/generation/Utilities.groovy]: Can this be more specific?  I think this is going to catch some tests?
Also don't lead with .*

Comment (jaredpar) [inline jobs/generation/Utilities.groovy]: I'd avoid this particular regex.  The second one seems to catch all of the cases we've seen lately.  I agree this one is a bit broad and would catch a lot of extra cases.

File: jobs/generation/Utilities.groovy [hunk 2]
```diff
@@ -637,7 +637,9 @@ class Utilities {
...
+            'hexadecimal value.*is an invalid character.' // This is here until NuGet cache corruption issue is root caused and fixed.
```
Comment (alexperovich) [inline jobs/generation/Utilities.groovy]: Can groovy files have trailing commas in these initializers? If so there should probably be a trailing comma here to make the diffs more clean when adding new entries.

Comment (mmitche) [inline jobs/generation/Utilities.groovy]: I believe they can.

File: jobs/generation/Utilities.groovy (hunk 1 above)
Comment (smile21prc) [inline jobs/generation/Utilities.groovy]: Thank you both for the inputs. I've removed the 1st regex and only left the 2nd one.

File: jobs/generation/Utilities.groovy (hunk 2 above)
Comment (smile21prc) [inline jobs/generation/Utilities.groovy]: According to http://docs.groovy-lang.org/next/html/documentation/working-with-collections.html, extra trailing comma seems to be ok in the case below, likely the same for initializer:
assert list == [5, 6, 9, 8,]           // trailing comma OK

I've added the trailing comma to the end to ease diffs. @mmitche , please let me know if this is against either grammar or coding convention in these groovy files.

**Comment by mafei071111** [comment]
mafei071111

**Pull request merged**